Export/Classes/GGPKFile.lua

```diff
@@ -61,19 +124,30 @@ function GGPKClass:ReadRecord(record)
 				offset = bytesToULong(raw, nameLength * 2 + (i-1) * 12 + 5),
 			}
 		end
---		ConPrintf("PDIR '%s': %d records", record.name, record.numRecord)
+		--ConPrintf("PDIR '%s': %d records", record.name, record.numRecord)
 	elseif record.tag == "FILE" then
 		raw = self.file:read(36)
 		local nameLength = bytesToUInt(raw, 1)
-		record.hash = raw:sub(5)
+		record.hash = raw:sub(5) -- 32 bytes starting at [4:36]
 		record.name = convertUTF16to8(self.file:read(nameLength * 2))
-		local headLength = 44 + nameLength * 2
+		local headLength = 44 + nameLength * 2 -- 44 = 36 read + (4 for rec_leng + 4 for type) pulled outside of 'if'
 		record.dataOffset = record.offset + headLength
 		record.dataLength = record.length - headLength
---		ConPrintf("FILE '%s': %d bytes", record.name, record.dataLength)
+		--ConPrintf("FILE '%s': %d bytes", record.name, record.dataLength)
+		--if self.dump then
+		is_needed = self:RecordNeeded(record.name)
+		if is_needed == 1 then
```

Comment:
I'd like to see something more expressive than numeric flags. Describe in the code what `isNeeded` being equal to 1 or 2 means.